llvm/lib/Transforms/InstCombine/InstCombineMulDivRem.cpp

```diff
@@ -1544,6 +1544,20 @@ Instruction *InstCombinerImpl::visitSDiv(BinaryOperator &I) {
     }
   }
 
+  // -(X * Y) / (X * Y) --> -1
+  if ((match(Op0, m_Neg(m_c_Mul(m_Value(X), m_Value(Y)))) &&
+       match(Op1, m_c_Mul(m_Specific(X), m_Specific(Y)))) ||
+      (match(Op0, m_c_Mul(m_Value(X), m_Value(Y))) &&
+       match(Op1, m_Neg(m_c_Mul(m_Specific(X), m_Specific(Y)))))) {
```

Comment:
```suggestion
  if ((match(Op0, m_NSWNeg(m_c_Mul(m_Value(X), m_Value(Y)))) &&
       match(Op1, m_c_Mul(m_Specific(X), m_Specific(Y)))) ||
      (match(Op0, m_Mul(m_Value(X), m_Value(Y))) &&
       match(Op1, m_NSWNeg(m_c_Mul(m_Specific(X), m_Specific(Y)))))) {
```

Reply:
You could also:
1) check knownbits to see that X/Y are not INT_MIN
2) You could also handle `((-a) * b) / (a * b)` (which doesn't get canonicalized to `(-(a * b)) / (a * b)`)